include/vulkan/vk_platform.h

```diff
@@ -85,6 +85,12 @@ extern "C"
     #endif
 #endif // !defined(VK_NO_STDINT_H)
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+    // Forward declared for VK_EXT_acquire_wl_display
+    struct zwp_drm_lease_manager_v1;
+    struct zwp_drm_lease_connector_v1;
+#endif // VK_USE_PLATFORM_WAYLAND_KHR
```

Comment:
Hmm, so far we have been including stuff directly with headers. Though TBH, I would prefer your way, instead of dragging in unnecessary header dependencies.

Reply:
Yes, I prefer forward declarations as well when feasible.

Reply:
Nonetheless this kinda circumvents the `vk.xml` autogenerating system.

Reply:
If there's interest in adding features for forward-declaration of platform types, I'd be interested too on the OpenXR side. I am pretty sure it actually can be done: I think I managed to get an include added without changing any code, just xml.

Reply:
It was being "solved" a bit here, when I complained about dragging in the whole monolithic `windows.h`. The solution arrived at here is that `vulkan.h` is supposed to just include everything, and if you want to control `include`s you are supposed to be including the `vulkan_*.h` stuff directly.

Of course that does not exactly scale, as at some point the extensions will want to include the whole universe, making `vulkan.h` unusable. Not to mention given header might not even exist on the user's PC.

Anyway we are offtopicing a bit here. I suggest you start an Issue.
Anyway, a precedent is to create a new platform (e.g. `VK_USE_PLATFORM_XLIB_XRANDR_EXT`). AIS forward declarations are fine with me, then again this is not my repo, and my opinion does not matter.

Reply:
I don't have a strong opinion, but I recall the way I did it with XLIB_XRANDR caused some problems as well.  I don't remember exactly why.

In general, there was a lot of time pressure behind the VR-related extensions, and we didn't have as much time for discussion and especially finessing of these infrastructure/backend details as I would have liked, so stick to first principles when deciding whether to carry their design tropes forward.